tests/proxyname_test.go

```diff
@@ -78,55 +82,70 @@ func TestProxyNameLabel(t *testing.T) { // nolint
 			// label the disabled service
 			mustOrFatal(disabledService.SetLabel(labelKey, labelValue), t)
 			return ctx
-		}).
-		Teardown(func(context.Context, *testing.T, *envconf.Config) context.Context {
-			services := []*kubernetes.Service{disabledService.(*kubernetes.Service), toggledService.(*kubernetes.Service)}
-			tools.ResetTestBoard(t, services, model)
-			return ctx
-		}).
-		Assess("should implement service.kubernetes.io/service-proxy-name", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
-			ma.Logger.Info("verify the toggledServices are up.")
+		}
+	}
+
+	teardown := func(context.Context, *testing.T, *envconf.Config) context.Context {
+		for _, service := range []kubernetes.ServiceBase{disabledService, toggledService} {
+			if err := service.Delete(); err != nil {
+				t.Fatal(err)
+			}
+		}
+		return ctx
+	}
+
+	getAssessFunc := func(labelKey string) features.Func {
+		return func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			ma.Logger.Info("verify the toggledService is up without", zap.String("label", labelKey))
 			toPod.SetClusterIP(toggledClusterIP)
 			tools.MustNoWrong(matrix.ValidateOrFail(ma, model, &matrix.TestCase{
 				ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: upReachability,
 				ServiceType: entities.ClusterIP,
 			}, false), t)
 
-			ma.Logger.Info("verify the disabledServices are not up.")
+			ma.Logger.Info("verify the disabledService is not up with", zap.String("label", labelKey))
 			toPod.SetClusterIP(disabledClusterIP)
 			tools.MustNoWrong(matrix.ValidateOrFail(ma, model, &matrix.TestCase{
 				ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: downReachability,
 				ServiceType: entities.ClusterIP,
 			}, false), t)
 
-			ma.Logger.Info("add service-proxy-name label.")
-			mustOrFatal(toggledService.SetLabel(labelKey, labelValue), t)
+			ma.Logger.Info("add label to the toggledService", zap.String("label", labelKey))
+			mustOrFatal(toggledService.SetLabel(proxyNameLabelKey, labelValue), t)
```

Comment:
should use `labelKey` instead of `proxyNameLabelKey`

Reply:
wow good catch!